sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
@isuruf we can have both in symengine.py, we just need to implement this in the wrappers (you are right that in C++ we can't do that). I don't now if we should, since things will get inevitably more complex.

We can however write a compatible layer on top of the native symengine.py, or use the approach described in https://github.com/symengine/symengine/wiki/SymPy-core-upgrade-to-SymEngine.